core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -139,15 +139,10 @@ trait WorkflowComponent {
           marshalInputResolution(inputResolution, workflowRecsByEntity(workflow.workflowEntity).id)
         }
 
-        val insertedRecQuery = for {
-          workflowRec <- findWorkflowsBySubmissionId(submissionId)
-          workflowEntityRec <- entityQuery if workflowEntityRec.id === workflowRec.workflowEntityId
-          insertedInputResolutionRec <- submissionValidationQuery if insertedInputResolutionRec.workflowId === workflowRec.id
-        } yield (workflowEntityRec, insertedInputResolutionRec)
+        val insertedRecQuery = submissionQuery.loadSubmissionValidationsAndReferences(submissionId)
 
         insertInBatches(submissionValidationQuery, inputResolutionRecs) andThen
-        insertedRecQuery.result.map(_.map { case (workflowEntityRec, insertedInputResolutionRec) =>
```

Comment:
Since we're going from Query to Action here, it makes sense that Slick would return a giant pile of data before we trimmed it on the app side.  I wonder if Slick would be smart enough to do the data reduction for us if we moved this logic inside the Query.

Great find!